src/Histogram.hpp

```diff
@@ -43,4 +48,34 @@ class shasta::Histogram : public vector<size_t> {
 };
 
 
+// A histogram class that calculates bin size and finds the appropriate bin to increment given an x value
+class shasta::IterativeHistogram{
+public:
+    /// Attributes ///
+    const double start;
+    const double stop;
+    const size_t nBins;
+    const double binSize;
+    vector<uint64_t> histogram;
+    bool unboundedLeft;
+    bool unboundedRight;
+
+    /// Methods ///
+    IterativeHistogram(
+            double start,
+            double stop,
+            size_t nBins,
```

Comment:
Calling this `binCount` instead of `nBins` would be more consistent with typical naming in Shasta code.